protocols/grpc/agent.proto

```diff
@@ -55,6 +55,7 @@ service AgentService {
 	rpc OnlineCPUMem(OnlineCPUMemRequest) returns (google.protobuf.Empty);
 	rpc ReseedRandomDev(ReseedRandomDevRequest) returns (google.protobuf.Empty);
 	rpc GetGuestDetails(GuestDetailsRequest) returns (GuestDetailsResponse);
+	rpc UploadFile(UploadFileRequest) returns (google.protobuf.Empty);
```

Comment:
I have a global question here, it feels like having only `UploadFile()` here is only one part of the solution as it would not allow for the update of those files. I think we should, as part of this PR, make the update being supported too.
I'm not against having both initial upload and update being supported by the same function, but it should be explicit if we go that way, otherwise we can simply have two different functions.

Reply:
what is going to be the difference between upload and update function? upload can be used to update a file

Reply:
Sorry, wouldn’t doing a second uploadFile call overwrite the first?

Reply:
I was thinking the same, the caller would call same again to update.

Reply:
I'm not saying it's not working (the way it's implemented makes it work) but the naming does not sound explicit to me that you should reuse the same function to update an existing file.

Reply:
sigh, what name sounds explicit to you?

Reply:
We could call it CopyFile, wdyt?

Reply:
I don't have a strong opinion either way.  I'll poll some other kata folks around me and provide suggestion :)